File: hw/cevallos/u2/hw19FarmSystem/Python/Sheep.py
from FarmAnimal import FarmAnimal

class Sheep(FarmAnimal):
    def __init__(self, last_sheering, id, breed, born_on, gender, is_able_to_reproduce, weight, cage):
        super().__init__(id, breed, born_on, gender, is_able_to_reproduce, weight, cage)
        self.last_sheering = last_sheering
    
    def cut_wool(self):
        pass
    
    def shear(self):
        pass
    
    def __str__(self):
        return f"Sheep{{lastSheering={self.last_sheering}, {super().__str__()}}}"

File: hw/cevallos/u2/hw19FarmSystem/JavaScript/Sheep.js
const FarmAnimal = require('./FarmAnimal');

class Sheep extends FarmAnimal {
    constructor(lastSheering, id, breed, bornOn, gender, isAbleToReproduce, weight, cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.lastSheering = lastSheering;
    }

    cutWool() {
        // Implementation for cutting wool
    }

    shear() {
        // Implementation for shearing
    }

    toString() {
        return `Sheep{lastSheering=${this.lastSheering}, ${super.toString()}}`;
    }
}

module.exports = Sheep;